### DISCUSSION TITLE
How to resolve ts-check "class" JSX attribute issues?

### DISCUSSION THREAD

**Discussion by zgoda** [preactjs/preact#3131] [Q&A]
I have a plain JS code that I'd like to have type-checked with TS. Adding `// @ts-check` to module source is enough in VS Code and it works great except for that `class` JSX attribute that is allowed by Preact although it's not defined in JSX typings and therefore reported as error. I tried a [solution from this article](https://dev.to/swyx/how-to-use-class-instead-of-classname-with-preact-and-typescript-2bjh) but it does not work for me because it's plain JS so the TS checker is run by VSC independently of project config. I suppose this may be unsolvable on my side because it's internal TS checker but maybe someone tried that before or has any idea?

**Comment by wg-daniel** [comment]
@zgoda Long shot, but where you able to resolve this in the end?

**Reply by zgoda** [reply]
As a workaround I'm using TS config from WMR based project but this enables TS checks globally. I don't consider this to be a solution but it works for my projects.

**Reply by zgoda** [reply]
As a workaround I'm using TS config from WMR based project but this enables TS checks globally. I don't consider this to be a solution but it works for my projects.
